distributed/cli/tests/test_dask_worker.py

```diff
@@ -682,3 +683,45 @@ def dask_setup(worker):
         await c.wait_for_workers(1)
         [foo] = (await c.run(lambda dask_worker: dask_worker.foo)).values()
         assert foo == "setup"
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize("nanny", ["--nanny", "--no-nanny"])
+def test_timeout(nanny):
+    with popen(
+        ["dask-worker", "192.168.1.100:7777", nanny, "--death-timeout=1"],
+        flush_output=False,
+    ) as worker:
+        logs = [worker.stdout.readline().decode().lower() for _ in range(15)]
+
```

Comment:
I think you can avoid using `with popen` here and use `subprocess.run(["dask-worker", "192.168.1.100:7777", nanny, "--death-timeout=1"], text=True, encoding="utf8", capture_output=True)`